functions/user-api/src/impl/user-api-impl.ts

```diff
@@ -0,0 +1,112 @@
+import {CallableRequest, HttpsError} from "firebase-functions/v2/https";
+import * as logger from "firebase-functions/logger";
+import {Datastore, Entity, PropertyFilter} from "@google-cloud/datastore";
+
+const DATASTORE_USER_INFORMATION_KIND = "web_api_users";
+const UID_PROPERTY_NAME = "uid";
+
+export const retrieveUser = async (
+  datastore: Datastore,
+  uid: string):
+  Promise<Entity | undefined> => {
+  /**
+   * Retrieve the user information from datastore
+   * @param datastore - Datastore object
+   * @param uid - The user's uid
+   */
```

Comment:
Applicable to all function comments, add the comment block outside the function definition.